plugins/bcc-login/includes/class-bcc-login-visibility.php

```diff
@@ -323,13 +322,22 @@ function on_block_editor_assets() {
      * @return WP_Query
      */
     function filter_pre_get_posts( $query ) {
+        // Don't filter posts for Phrase
+        // Check if Phrase (Memsource) is installed
+        if ( class_exists('\Memsource\Utils\AuthUtils') ) {
+            // Validate the token in the request
+            if ( \Memsource\Utils\AuthUtils::validateTokenInRequest() ) {
```

Comment:
Is there a "cheaper" way of checking if the token exists before trying to validate it. This will probably get run on every request.

Reply:
What we can check before everything is whether 'token' is present in the $_GET array, but I'm not sure if that's a cheaper way. The validate function is not very "expensive". Have a look here:

![image](https://github.com/user-attachments/assets/6d6be500-0666-4f13-90fb-bd870b966421)

Reply:
I would check getTokenFromRequest() first, and then validate. It will save a trip to the DB in many instances - unless i'm missing something here.